packages/aws-cdk/lib/commands/flag-operations.ts

```diff
@@ -41,456 +44,643 @@ interface FlagOperationsParams {
-    all: options.all,
-    value: options.value,
-    flagName: options.FLAGNAME,
-    default: options.default,
-    unconfigured: options.unconfigured,
-  };
-
-  const interactiveOptions = Object.values(FlagsMenuOptions);
-
-  if (options.interactive) {
-    const prompt = new Select({
-      name: 'option',
-      message: 'Menu',
-      choices: interactiveOptions,
-    });
+  /**
+   * Main entry point for handling flag operations based on user options.
+   */
+  public async executeFlagOperations(): Promise<void> {
+    if (this.flags.length == 0) {
+      await this.ioHelper.defaults.error('The \'cdk flags\' command is not compatible with the AWS CDK library used by your application. Please upgrade to 2.212.0 or above.');
+      return;
+    }
 
-    const answer = await prompt.run();
-    if (answer == FlagsMenuOptions.ALL_TO_RECOMMENDED) {
-      params = {
-        ...params,
-        recommended: true,
-        all: true,
-      };
-      await setMultipleFlags(params);
-    } else if (answer == FlagsMenuOptions.UNCONFIGURED_TO_RECOMMENDED) {
-      params = {
-        ...params,
-        recommended: true,
-        unconfigured: true,
-      };
-      await setMultipleFlags(params);
-    } else if (answer == FlagsMenuOptions.UNCONFIGURED_TO_DEFAULT) {
-      params = {
-        ...params,
-        default: true,
-        unconfigured: true,
-      };
-      await setMultipleFlagsIfSupported(params);
-    } else if (answer == FlagsMenuOptions.MODIFY_SPECIFIC_FLAG) {
-      await setFlag(params, true);
-    } else if (answer == FlagsMenuOptions.EXIT) {
+    const interactiveOptions = Object.values(FlagsMenuOptions);
+
+    if (this.params.interactive == true) {
+      const prompt = new Select({
+        name: 'option',
+        message: 'Menu',
+        choices: interactiveOptions,
+      });
+
+      const answer = await prompt.run();
+      if (answer == FlagsMenuOptions.ALL_TO_RECOMMENDED) {
+        this.params.recommended = true;
+        this.params.all = true;
+        await this.setMultipleFlags();
+      } else if (answer == FlagsMenuOptions.UNCONFIGURED_TO_RECOMMENDED) {
+        this.params.recommended = true;
+        this.params.unconfigured = true;
+        await this.setMultipleFlags();
+      } else if (answer == FlagsMenuOptions.UNCONFIGURED_TO_DEFAULT) {
+        this.params.default = true;
+        this.params.unconfigured = true;
+        await this.setMultipleFlagsIfSupported();
+      } else if (answer == FlagsMenuOptions.MODIFY_SPECIFIC_FLAG) {
+        await this.setFlag(true);
+      } else if (answer == FlagsMenuOptions.EXIT) {
+        return;
+      }
       return;
     }
-    return;
-  }
 
-  if (options.FLAGNAME && options.all) {
-    await ioHelper.defaults.error('Error: Cannot use both --all and a specific flag name. Please use either --all to show all flags or specify a single flag name.');
-    return;
-  }
+    if (this.params.safe) {
+      await this.setSafeFlags();
```

Comment:
Wait we did all the sets already in the big `if/else` chain above. Isn't this too late?